src/CFamily/SystemAbstractions/IProcess.cs

```diff
@@ -0,0 +1,107 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+
+/// <summary>
+/// Wrappers around System.Diagnostics.Process for testing
+/// </summary>
+
+namespace SonarLint.VisualStudio.CFamily.SystemAbstractions
+{
+    internal interface IProcessFactory
+    {
+        IProcess Start(ProcessStartInfo startInfo);
+    }
+
+    internal interface IProcess : IDisposable
+    {
+        int Id { get; }
+        bool HasExited { get; }
+        int ExitCode { get; }
+        ProcessStartInfo StartInfo { get; }
```

Comment:
The `StartInfo` property isn't used by the `VSDevCmd` runner. It's there mainly to help with testing the factory class i.e. to make sure it has passed through the expected instance.